mmocr/datasets/preparers/data_preparer.py

```diff
@@ -115,12 +119,38 @@ def parse_cfg(self, cfg_path: str) -> None:
         if 'data_converter' in cfg:
             cfg.data_converter.update(
                 dict(nproc=self.nproc, dataset_name=self.dataset_name))
+            if self.dump_to_lmdb:
+                cfg.data_converter.update(dict(dumper=dict(type='LMDBDumper')))
             self.data_converter = DATA_CONVERTERS.build(cfg.data_converter)
         if 'config_generator' in cfg:
             cfg.config_generator.update(
                 dict(
                     dataset_name=self.dataset_name,
                     overwrite_cfg=self.overwrite_cfg))
+            if self.dump_to_lmdb:
+                anns_dict = {}
+                for ann_list_key in ['train_anns', 'val_anns', 'test_anns']:
+                    if ann_list_key in cfg.config_generator:
+                        ann_list = cfg.config_generator[ann_list_key]
+                        for ann_dict in ann_list:
+                            suffix = ann_dict['ann_file'].split('.')[-1]
+                            lmdb_ann_file = ann_dict['ann_file'].replace(
+                                suffix, 'lmdb')
```

Comment:
```suggestion
                            name_chunks = ann_dict['ann_file'].split('.')
                            name_chunks[-1] = 'lmdb'
                            lmdb_ann_file = '.'.join(name_chunks)
```
Using replace may unintentionally change the filename (e.g. `xxxjsonxx.json` would be changed to `xxxlmdbxx.lmdb`)

Reply:
Done, I have corrected my mistake.